training.m

```diff
@@ -15,6 +15,16 @@
             F(3) = (1 - a1(3))*a1(3);
             F(4) = (1 - a1(4))*a1(4);
             F(5) = (1 - a1(5))*a1(5);
+            F(6) = (1 - a1(6))*a1(6);
+            F(7) = (1 - a1(7))*a1(7);
+            F(8) = (1 - a1(8))*a1(8);
+            F(9) = (1 - a1(9))*a1(9);
+            F(10) = (1 - a1(10))*a1(10);
+            F(11) = (1 - a1(11))*a1(11);
+            F(12) = (1 - a1(12))*a1(12);
+            F(13) = (1 - a1(13))*a1(13);
+            F(14) = (1 - a1(14))*a1(14);
+            F(15) = (1 - a1(15))*a1(15);
```

Comment:
This could be improved by an external function to help "build" that matrix instead of these `F(n)`